osu.Game/Beatmaps/BeatmapDifficultyManager.cs

```diff
@@ -114,6 +115,27 @@ public StarDifficulty GetDifficulty([NotNull] BeatmapInfo beatmapInfo, [CanBeNul
             return computeDifficulty(key, beatmapInfo, rulesetInfo);
         }
 
+        /// <summary>
+        /// Calculates performance for the given <see cref="ScoreInfo"/> on a given <see cref="WorkingBeatmap"/>.
+        /// </summary>
+        /// <param name="beatmap">The <see cref="WorkingBeatmap"/> to do the calculation on. </param>
+        /// <param name="score">The score to do the calculation on. </param>
+        /// <param name="token">An optional <see cref="CancellationToken"/> to cancel the operation.</param>
+        /// <returns></returns>
+        public async Task<double> CalculatePerformance([NotNull] WorkingBeatmap beatmap, [NotNull] ScoreInfo score, CancellationToken token = default)
```

Comment:
the lack of caching on this is concerning, especially seeing that this seems to be called three times on entering the results screen and once more on every time you expand another score and then go back to the original one

Reply:
Seeing that the expanded part of the first score panel (I meant by first the panel you land on when you access results) is constructed 3 times in a row, it may be a good idea as you said to cache calculation somewhere (and move the calculation there aswell), maybe in `ScorePanel` (since it doesn't get reconstructed each time you select another score) as adding a cache lookup to the `DifficultyManager` feels overkill to me

Reply:
overkill why? is there a reason to ever recalculate a hashed score's PP after it's already been computed once?

by the way I missed this entirely yesterday but the fact that performance calculation is happening in `BeatmapDifficultyManager` seems suboptimal from a code organisation standpoint. the class is supposed to be doing things related to difficulty, so it should probably be split.